src/WingetCreateCLI/Commands/UpdateCommand.cs

```diff
@@ -419,6 +443,54 @@ private static void DisplayManifestsAsMenuSelection(Manifests manifests)
             }
         }
 
+        /// <summary>
+        /// Parse out architecture overrides included in the installer URLs and returns the parsed list of installer URLs.
+        /// </summary>
+        /// <param name="installerUrlsToBeParsed">List of installer URLs to be parsed for architecture overrides.</param>
+        /// <returns>List of <see cref="InstallerUpdateHelper"/> helper objects used for updating the installers.</returns>
+        private List<InstallerUpdateHelper> ParseInstallerUrlsForArchOverride(List<string> installerUrlsToBeParsed)
+        {
+            List<InstallerUpdateHelper> installerUpdateList = new List<InstallerUpdateHelper>();
+            foreach (string item in installerUrlsToBeParsed)
+            {
+                InstallerUpdateHelper installerUpdate = new InstallerUpdateHelper();
+
+                if (item.Contains('|'))
+                {
+                    // '|' character indicates that an architecture override can be parsed from the installer.
+                    string[] installerUrlOverride = item.Split('|');
+
+                    if (installerUrlOverride.Length > 2)
+                    {
+                        Logger.ErrorLocalized(nameof(Resources.MultipleArchitectureOverride_Error));
+                        return null;
+                    }
+
+                    string installerUrl = installerUrlOverride[0];
+                    string overrideArchString = installerUrlOverride[1];
+                    InstallerArchitecture? overrideArch = overrideArchString.ToEnumOrDefault<InstallerArchitecture>();
+                    if (overrideArch.HasValue)
+                    {
+                        installerUpdate.InstallerUrl = installerUrl;
+                        installerUpdate.OverrideArchitecture = overrideArch.Value;
+                        installerUpdateList.Add(installerUpdate);

```

Comment:
Factor out of the if blocks

Reply:
factored out of if blocks

Reply:
I meant the larger if blocks :)

Reply:
ah i see, factored out of larger if block